Automattic/woocommerce-payments
# Add payment processing using ECE on the Blocks checkout and cart pages

Closes #8773 
Closes #8867 

#### Changes proposed in this Pull Request

Add payment processing using Express Checkout Elements (ECE) on the Blocks checkout and cart pages.

#### Testing instructions

**Test that there are no regressions in Payment Request Buttons (PRB) when ECE is disabled:**

Pre-requisites:
- The feature flag is disabled (`npm run wp option update _wcpay_feature_stripe_ece 0`).
- Confirm that Apple Pay and Google Pay are enabled in the merchant settings.

Test:

1. [Checkout with Payment Request](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#checkout-with-payment-request)

**Test the ECE checkout on the blocks cart page using a simple card:**

Pre-requisites:
- The feature flag is enabled (`npm run wp option update _wcpay_feature_stripe_ece 1`).
- Confirm that Apple Pay and Google Pay are enabled in the merchant settings.

Test:

1. Add some products to the cart.
2. Navigate to the blocks cart page and wait for the ECE button to load.
3. Complete the payment using the ECE button via Google Pay. Use a real card; however, your card will not be charged if the merchant is in test mode.
4. Ensure the payment is processed successfully and that you are redirected to the thank you page.
5. Review the order in the merchant backend and ensure all the data is correct.
6. Check the order details in the Stripe dashboard to ensure all information is correct.

**Test the ECE checkout in the blocks checkout page using a simple card:**

Pre-requisites:
- The feature flag is enabled (`npm run wp option update _wcpay_feature_stripe_ece 1`).
- Confirm that Apple Pay and Google Pay are enabled in the merchant settings.

Test:

1. Add some products to the cart.
2. Navigate to the blocks checkout page and wait for the ECE button to load.
3. Complete the payment using the ECE button via Google Pay. Use a real card; however, your card will not be charged if the merchant is in test mode.
4. Ensure the payment is processed successfully and that you are redirected to the thank you page.
5. Review the order in the merchant backend and ensure all the data is correct.
6. Check the order details in the Stripe dashboard to ensure all information is correct.

**Ensure that there are no regressions when no express checkout payment methods are active.**

Pre-requisites:
- The feature flag is enabled (`npm run wp option update _wcpay_feature_stripe_ece 1`).
- Disable all express checkout payment methods in the merchant settings.

Test:

1. Add some products to the cart.
2. Navigate to the blocks checkout page, blocks cart page, shortcode checkout page, shortcode cart page, and product pages. Confirm that no express checkout buttons are visible.
3. Complete a checkout using a card on the classic checkout. Verify that the checkout process is completed without any errors.
4. Repeat step 1 by adding some products to the cart.
5. Complete a checkout using a card on the blocks checkout. Ensure the checkout process is completed without any errors.

-------------------


**Post merge**
